component/main.jsonnet

```
@@ -5,6 +5,161 @@ local inv = kap.inventory();
+    },
+    labels+: {
+      'app.kubernetes.io/name': params.namespace,
+      'topolvm.cybozu.com/webhook': 'ignore',
+    },
+  },
+};
+
+local lvmd_config = {
+  'socket-name': '/run/topolvm/lvmd.sock',
+  'device-classes': [
+    params.deviceclasses[name] { name: name }
+    for name in std.objectFields(params.deviceclasses)
+  ],
+};
+
+local lvmd_configmap = kube.ConfigMap('topolvm-lvmd0') {
+  metadata+: {
+    labels+: {
+      idx: '0',
+      'app.kubernetes.io/name': 'topolvm',
+      'app.kubernetes.io/instance': 'topolvm',
+      'app.kubernetes.io/version': params.images.topolvm.tag,
+    },
+  },
+  data: {
+    'lvmd.yaml': std.manifestYamlDoc(lvmd_config),
+  },
+};
+
+local lvmd_daemonset = kube.DaemonSet('topolvm-lvmd0') {
+  metadata+: {
+    labels+: {
+      idx: '0',
+      'app.kubernetes.io/name': 'topolvm',
+      'app.kubernetes.io/instance': 'topolvm',
+      'app.kubernetes.io/version': params.helmValues.image.tag,
+    },
+  },
+  spec: {
+    selector: {
+      matchLabels: {
+        idx: '0',
+        'app.kubernetes.io/name': 'topolvm-lvmd0',
+      },
+    },
+    template: {
+      metadata: {
+        labels: {
+          idx: '0',
+          'app.kubernetes.io/name': 'topolvm-lvmd0',
+        },
+        annotations: {
+          'prometheus.io/port': '8080',
+        },
+      },
+      spec: {
+        serviceAccountName: 'topolvm-lvmd',
+        hostPID: true,
+        containers: [
+          {
+            name: 'lvmd',
+            image: params.images.topolvm.registry + '/' + params.images.topolvm.repository + ':' + params.images.topolvm.tag,
+            securityContext: {
+              privileged: true,
+            },
+            command: [
+              '/lvmd',
+              '--container',
+            ],
+            resources: {
+              limits: {
+                cpu: params.resources.lvmd.limits.cpu,
+                memory: params.resources.lvmd.limits.memory,
+              },
+              requests: {
+                cpu: params.resources.lvmd.requests.cpu,
+                memory: params.resources.lvmd.requests.memory,
+              },
+            },
+            volumeMounts: [
+              {
+                name: 'lvmd-socket-dir',
+                mountPath: '/run/topolvm',
+              },
+              {
+                name: 'config',
+                mountPath: '/etc/topolvm',
+              },
+            ],
+          },
+        ],
+        volumes: [
+          {
+            name: 'config',
+            configMap: {
+              name: 'topolvm-lvmd0',
+            },
+          },
+          {
+            name: 'lvmd-socket-dir',
+            hostPath: {
+              path: '/run/topolvm',
+              type: 'DirectoryOrCreate',
+            },
+          },
+        ],
+        nodeSelector: {
+          'syn.tools/topolvm': '',
+        },
+      },
+    },
+  },
+};
+
+local StorageClass(name='ssd-local') = kube.StorageClass(name) {
```

Please use the component library of component `storageclass`.

```suggestion
local sc = import 'lib/storageclass.libsonnet';
local StorageClass(name='ssd-local') = sc.StorageClass(name) {
```